shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/RewardProvenance.hs

```diff
@@ -0,0 +1,180 @@
+{-# LANGUAGE BangPatterns #-}
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE DeriveAnyClass #-}
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE StandaloneDeriving #-}
+{-# LANGUAGE UndecidableInstances #-}
+{-# OPTIONS_GHC -Wno-orphans #-}
+
+module Shelley.Spec.Ledger.RewardProvenance where
+
+import qualified Cardano.Ledger.Crypto as CC
+import Control.DeepSeq (NFData)
+import Data.Aeson
+import Data.Default.Class (Default (..))
+import Data.Map.Strict (Map)
+import Data.Word (Word64)
+import GHC.Generics
+import NoThunks.Class (NoThunks (..))
+import Numeric.Natural (Natural)
+import Shelley.Spec.Ledger.Coin (Coin (..))
+import Shelley.Spec.Ledger.Credential (Credential (..))
+import Shelley.Spec.Ledger.EpochBoundary (BlocksMade (..))
+import Shelley.Spec.Ledger.Keys (KeyHash (..), KeyRole (..))
+import Shelley.Spec.Ledger.Orphans ()
+import Shelley.Spec.Ledger.TxBody (PoolParams (..), RewardAcnt (..))
+
+-- instances only
+-- ==========================================================
+
+data RewardProvenancePool crypto = RewardProvenancePool
+  { poolBlocksP :: !Natural,
+    sigmaP :: !Rational,
+    sigmaAP :: !Rational,
+    ownerStakeP :: !Coin,
+    poolParamsP :: !(PoolParams crypto),
+    pledgeRatioP :: !Rational,
+    maxPP :: !Coin,
+    appPerfP :: !Rational,
+    poolRP :: !Coin,
+    --mRewardsP :: !(Map (Credential 'Staking crypto) Coin),
+    lRewardP :: !Coin
+  }
+  deriving (Eq, Generic)
+
+instance NoThunks (RewardProvenancePool crypto)
+
+instance NFData (RewardProvenancePool crypto)
+
+deriving instance (CC.Crypto crypto) => FromJSON (RewardProvenancePool crypto)
+
+deriving instance (CC.Crypto crypto) => ToJSON (RewardProvenancePool crypto)
+
+instance Default (RewardProvenancePool crypto) where
+  def = RewardProvenancePool 0 0 0 (Coin 0) def 0 (Coin 0) 0 (Coin 0) (Coin 0)
+
+data RewardProvenance crypto = RewardProvenance
+  { spe :: !Word64,
+    blocks :: !(BlocksMade crypto),
+    maxLL :: !Coin,
+    deltaR1 :: !Coin,
+    deltaR2 :: !Coin,
+    r :: !Coin,
+    totalStake :: !Coin,
+    blocksCount :: !Integer,
+    d :: !Rational,
+    expBlocks :: !Integer,
+    eta :: !Rational,
+    rPot :: !Coin,
+    deltaT1 :: !Coin,
+    activeStake :: !Coin,
+    pools ::
+      !( Map
+           (KeyHash 'StakePool crypto)
+           (RewardProvenancePool crypto)
+       ),
+    hitRateEstimates :: !(Map (KeyHash 'StakePool crypto) (Double, Double))
+    -- key -> (LiklihoodEstimate,Desireability)
```

Comment:
maybe say that's explaining the two doubles?

Reply:
Done.